tests/integration/golang/helpers/test.go

```diff
@@ -85,6 +95,80 @@ func (s *BaseTestSuite) SetupTest() {
 	s.Require().Nil(err)
 	s.TagFixtures = tagFixtures
 
-	// by default, unload everything.
-	s.Require().Nil(s.NamespaceFixtures.UnloadFixtures())
+	s.AddSetupHook(s.setup)
+	s.AddTearDownHook(s.tearDown)
+}
+
+func (s *BaseTestSuite) setup() {
+	s.Require().Nil(s.NamespaceFixtures.TruncateTables())
+
+	if !s.SkipCreateDefaultNamespace {
+		var err error
+		s.DefaultNamespace, err = s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+			Code:                "default",
+			DefaultExperimentID: common.GetPointer(int32(0)),
+		})
+		s.Require().Nil(err)
+
+		if !s.SkipCreateDefaultExperiment {
+			s.DefaultExperiment, err = s.ExperimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+				Name:           "Default",
+				LifecycleStage: models.LifecycleStageActive,
+				NamespaceID:    s.DefaultNamespace.ID,
+			})
+			s.Require().Nil(err)
+
+			s.DefaultNamespace.DefaultExperimentID = s.DefaultExperiment.ID
+			_, err = s.NamespaceFixtures.UpdateNamespace(context.Background(), s.DefaultNamespace)
+			s.Require().Nil(err)
+		}
+	}
+}
+
+func (s *BaseTestSuite) AddSetupHook(hook func()) {
+	s.setupHooks = append(s.setupHooks, hook)
+}
+
+func (s *BaseTestSuite) runSetupHooks() {
+	for _, hook := range s.setupHooks {
+		hook()
+	}
+}
+
+func (s *BaseTestSuite) SetupTest() {
+	if !s.ResetOnSubTest {
+		s.runSetupHooks()
+	}
+}
+
+func (s *BaseTestSuite) SetupSubTest() {
+	if s.ResetOnSubTest {
+		s.runSetupHooks()
+	}
+}
+
+func (s *BaseTestSuite) AddTearDownHook(hook func()) {
+	s.tearDownHooks = append([]func(){hook}, s.tearDownHooks...)
```

Comment:
why just not -> `s.tearDownHooks = append(s.tearDownHooks, hook)`

Reply:
because I would like this list to be LIFO, like the way `defer` works

Reply:
I mean we have no dependency on ordering of `tearDownHooks` it seems like. at least before.

Reply:
I think we will once we introduce the server, which is why I implemented it that way

Reply:
ok, let's see then!